src/components/chips/chips.spec.js

```diff
@@ -213,20 +213,40 @@ describe('<md-chips>', function() {
         expect(scope.removeChip.calls.mostRecent().args[1]).toBe(0);       // Index
       });
 
+      var CHIP_COMPONENT_TEMPLATE =
+        '<md-chips ng-model="items" md-on-chip-select="onChipSelect($index)" md-on-chip-deselect="onChipDeselect($index)"></md-chips>';
 
-      it('should call the select method when selecting a chip', function() {
-        var element = buildChips(CHIP_SELECT_TEMPLATE);
+      it("should call the select and deselect expressions", function(done) {
+        var element = buildChips(CHIP_COMPONENT_TEMPLATE);
         var ctrl = element.controller('mdChips');
+        var scope = element.scope();
+        var state = 0;
+
+        scope.onChipSelect = function(index) {
+          if (state == 0) {
+            expect(index).toBe(0);
+            state++;
+
+            ctrl.selectedChip = 1;
+          } else if (state == 1) {
+            expect(index).toBe(1);
+            done();
+          }
+        };
 
-        scope.selectChip = jasmine.createSpy('selectChip');
+        scope.onChipDeselect = function(index) {
+          if (state == 0) {
+            expect(index).toBe(-1);
+          } else if (state == 1) {
+            expect(index).toBe(0);
+          }
+        };
 
-        element.scope().$apply(function() {
-          ctrl.items = ['Grape'];
-          ctrl.selectChip(0);
-        });
 
-        expect(scope.selectChip).toHaveBeenCalled();
-        expect(scope.selectChip.calls.mostRecent().args[0]).toBe('Grape');
+        scope.$apply(function() {
+          ctrl.items = ['Grape', 'Banana'];
+          ctrl.selectedChip = 0;
```

Comment:
Why was this changed from `ctrl.selectChip(0)`?

In general, you should prefer using the method that is available (since it probably has additional logic), rather than manually setting the variable.